src/Microsoft.Identity.Web.DownstreamApi/DownstreamApi.cs

```diff
@@ -302,43 +303,65 @@ internal partial class DownstreamApi : IDownstreamApi
             CancellationToken cancellationToken = default)
         {
             // Downstream API URI
-            string apiUrl = effectiveOptions.GetApiUrl();
+            string apiUrl = effectiveOptions.GetApiUrl();
+
+            using HttpClient client = string.IsNullOrEmpty(serviceName) ? _httpClientFactory.CreateClient() : _httpClientFactory.CreateClient(serviceName);
 
-            // Creation of the Http Request message with the right HTTP Method
-            using HttpRequestMessage httpRequestMessage = new HttpRequestMessage(
-                new HttpMethod(effectiveOptions.HttpMethod),
+            async Task UpdateRequestAsync(HttpRequestMessage request)

```

Comment:
>Task [](http://example.com/codeflow?start=18&length=4)

Can you move this to outside the method? Also could probably return a `ValueTask` to save an allocation.

Reply:
Moved. Since this method will probably do an async network call to AAD or L2 cache, ValueTask will probably not help.